src/twisted/web/test/test_web.py

```diff
@@ -751,6 +751,41 @@ def test_retrieveNonExistentSession(self):
         self.assertNotIn(b"does-not-exist", request.cookies[0])
 
 
+    def test_getSessionExpired(self):
+        """
+        L{Request.getSession} generates a new session when the previous
+        session has expired.
+        """
+        clock = Clock()
+        site = server.Site(resource.Resource())
+        d = DummyChannel()
+        request = server.Request(d, 1)
+        request.site = site
+        request.sitepath = []
+
+        def SessionFactoryWithClock(site, uid):
+            """
+            Forward to normal session factory, but inject the clock.
+            """
+            session = OriginalSessionFactory(site, uid)
+            session._reactor = clock
```

Comment:
If this code doesn't know what `OriginalSessionFactory` is then it doesn't really know that that it's valid to set `_reactor` on it,